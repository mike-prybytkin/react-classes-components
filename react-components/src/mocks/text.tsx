const mockText = {
  homeLink: 'home',
  aboutLink: 'about',
  cardTitle: 'Fjallraven',
  searchPlaceholder: 'Search...',
  linkBackToHome: 'Back to home page',
  heading404: 'Error 404!',
  headingAboutUs: 'About Us!',
  headingMain: 'Welcome to our store!',
  pageNotReady: 'Sorry, page not ready yet...',
  loading: 'Loading ...',
  itemNotFound: 'Sorry, item not found...',
  headingForm: 'Create User!',
  notCreatedUserInForm: 'Sorry, no users have been created yet...',
};

export default mockText;
