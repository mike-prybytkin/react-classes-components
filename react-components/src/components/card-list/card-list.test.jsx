import React from 'react';
import { render, screen } from '@testing-library/react';
import CardList from './card-list';
import { mockCards } from '../../mocks/cards';

describe('Card list', () => {
  it('render without data', () => {
    render(<CardList cards={[]} />);

    expect(screen.queryByRole('heading')).toBeNull();
  });

  it('render cards', () => {
    render(<CardList cards={mockCards} />);

    expect(screen.getByText(/Fjallraven/i)).toBeInTheDocument();
    expect(screen.queryAllByRole('heading', { level: 3 })).toMatchSnapshot();
  });

  it('List snapshot', () => {
    const list = render(<CardList cards={mockCards} />);

    expect(list).toMatchSnapshot();
  });
});
