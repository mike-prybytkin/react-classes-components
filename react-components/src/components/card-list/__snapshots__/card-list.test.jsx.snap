// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card list List snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cards-wrapper"
      >
        <div
          class="card"
        >
          <h3
            class="card__title"
          >
            Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
          </h3>
          <img
            alt="men's clothing"
            class="card__img"
            src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
          />
          <p
            class="card__rating"
          >
            3.9
             ★
          </p>
          <p
            class="card__price"
          >
            109.95
             $
          </p>
        </div>
        <div
          class="card"
        >
          <h3
            class="card__title"
          >
            Mens Casual Premium Slim Fit T-Shirts 
          </h3>
          <img
            alt="men's clothing"
            class="card__img"
            src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
          />
          <p
            class="card__rating"
          >
            4.1
             ★
          </p>
          <p
            class="card__price"
          >
            22.3
             $
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cards-wrapper"
    >
      <div
        class="card"
      >
        <h3
          class="card__title"
        >
          Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
        </h3>
        <img
          alt="men's clothing"
          class="card__img"
          src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
        />
        <p
          class="card__rating"
        >
          3.9
           ★
        </p>
        <p
          class="card__price"
        >
          109.95
           $
        </p>
      </div>
      <div
        class="card"
      >
        <h3
          class="card__title"
        >
          Mens Casual Premium Slim Fit T-Shirts 
        </h3>
        <img
          alt="men's clothing"
          class="card__img"
          src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
        />
        <p
          class="card__rating"
        >
          4.1
           ★
        </p>
        <p
          class="card__price"
        >
          22.3
           $
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card list render cards 1`] = `
Array [
  <h3
    class="card__title"
  >
    Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
  </h3>,
  <h3
    class="card__title"
  >
    Mens Casual Premium Slim Fit T-Shirts 
  </h3>,
]
`;
